libs/search-widget/src/core/stores/answers.store.ts

```diff
@@ -1,13 +1,46 @@
+import { combineLatest } from 'rxjs';
+import type { Answer, DialogEntry } from '../answer.models';
 import { SvelteState } from '../state-lib';
 
-interface Answer {
-  answer: string;
-  sources: { resourceId: string; paragraph: string }; // TEMPORARY
-}
 interface AnswerState {
-  dialog: Answer[];
+  dialog: DialogEntry[];
+  currentAnswer: Answer;
+  isStreaming: boolean;
 }
 
+const EMPTY_ANSWER = { text: '', sources: [] };
 export const answerState = new SvelteState<AnswerState>({
   dialog: [],
+  currentAnswer: EMPTY_ANSWER,
+  isStreaming: false,
 });
+
+export const currentAnswer = answerState.writer<Partial<Answer>>(
```

Comment:
I don't understand why you set the writer as `Partial<Answer>` while you're returning `currentAnswer` which is of type `Answer`.
If you wanted to type the value sent to set the currentAnswer, I think you should have typed your writer as `answerState.writer<Answer, Partial<Answer>>`, no?